articles/server-platforms/aspnet-core/01-login.md

```diff
@@ -179,88 +93,71 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
-* [LockContext.cs](https://github.com/auth0-samples/auth0-aspnetcore-oidc-lock/blob/master/src/AspNetCoreOidcLockSample/LockContext.cs)
+For the Login you will need to return a `ChallengeResult` and specify "Auth0" as the authentication scheme which will be challenged. This will invoke the OIDC middleware you registered in the `Configure` method.
 
-::: panel-warning Fix the namespaces
-Be sure fix the namespaces in the downloaded files to correlate with those of your own project.
-:::
+After the OIDC middleware has signed the user in, the user will automatically be signed into the cookie middleware as well to authenticate them on subsequent requests. So for the `Logout` action you will need to sign the user out of both the OIDC and the cookie middleware: 
 
-### 8. Add an Account Controller
-
-Add an `AccountController` class:
-
-``` csharp
+```cs
 public class AccountController : Controller
 {
-    IOptions<OpenIdConnectOptions> _options;
-
-    public AccountController(IOptions<OpenIdConnectOptions> options)
-    {
-        _options = options;
-    }
-
-    // GET: /<controller>/
-    public IActionResult Login(string returnUrl = null)
+    public IActionResult Login()
     {
-        var lockContext = HttpContext.GenerateLockContext(_options.Value, returnUrl);
-
-        return View(lockContext);
+        return new ChallengeResult("Auth0", new AuthenticationProperties() { RedirectUri = "/" });
     }
 
-    public async Task<IActionResult> Logout()
+    public IActionResult Logout()
     {
-        // Sign the user out of the authentication middleware (i.e. it will clear the Auth cookie)
-        await HttpContext.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
+        HttpContext.Authentication.SignOutAsync("Auth0");
+        HttpContext.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
 
-        // Redirect the user to the home page after signing out
         return RedirectToAction("Index", "Home");
     }
 }
 ```
 
-The `Login` method will call the `GenerateLockContext` extension method which will create the correct `state` and `nonce` parameters and set the correct Cookies for the OIDC middleware to function correctly. It will return a `LockContext` parameter which you can pass along to the View.
-
-Also notice that the `OpenIdConnectOptions` are injected into the Controller, and that it is passed along to the `GenerateLockContext`. This method needs to retrieve the configuration settings for the OIDC middleware so that the correct `state` can be generated. It is for this reason why we had to register the `OpenIdConnectOptions` with the dependency injection.
-
-### 9. Setting up the View
-
-For the Login screen you can create a Razor view and embed the code for Lock. You can head over to our [Lock Library page](https://auth0.com/docs/libraries/lock#start-using-auth0lock) and confgure Lock the way you want, and then copy and paste the code it generated into your Login page.
```

Comment:
The text you referenced here is the deleted text.

I fixed it in the file 02-login-embedded-lock.md though